hubspot_sync/tasks.py

```diff
@@ -229,13 +232,12 @@ def batch_create_hubspot_objects_chunked(
             )
             for result in response.results:
                 if ct_model_name == "user":
-                    object_id = (
-                        User.objects.filter(
-                            email__iexact=result.properties["email"], is_active=True
-                        )
-                        .first()
-                        .id
-                    )
+                    user = User.objects.filter(
+                        email__iexact=result.properties["email"], is_active=True
+                    ).first()
+                    user.hubspot_sync_datetime = datetime.now()
+                    user.save()
+                    object_id = user.id
                 else:
                     object_id = result.properties["unique_app_id"].split("-")[-1]
                 HubspotObject.objects.update_or_create(
```

Comment:
I still kind of feel that the better place to store this might be in `HubspotObject`, because eventually we might run into the same problem if we need to update all existing deals in hubspot.  There aren't nearly as many deals as contacts (yet) but that might change once we start creating deals for free enrollments.

So could do this here:

```python
HubspotObject.objects.update_or_create(
                    content_type=content_type,
                    hubspot_id=result.id,
                    object_id=object_id,
                    defaults: {
                        "hubspot_sync_datetime": datetime.now()
                    }
                )
```

And also update the `mitol.hubspot_api.api.upsert_object_request` function:
```python
    if object_id:
        HubspotObject.objects.update_or_create(
            object_id=object_id, 
            content_type=content_type, 
            hubspot_id=hubspot_id or result.id, 
            defaults: {
                "hubspot_sync_datetime": datetime.now()
            }            
        )

Reply:
@mbertrand My other worry would be that updating applications other than MITx Online which directly call HubSpot would need to be prioritized for after this and the other MITx Online HubSpot work is completed, on June 12th.  I don't think that will cause problems, but there's the chance other work is brought forward prioritized ahead of this.

Reply:
It might be possible to move that direct bulk update call to ol-django too, at least partially.  I think the June 12th work involves adding all enrollments to hubspot as deals, and there are over 100K enrollments on prod, so it's possible we might have to do the same for those as for contacts to get them all created/updated.  But since the process of updating ol-django then adding the latest version to each app is cumbersome and 6/12 is fast approaching, I'll accept the current approach.  Will test functionality next.

Reply:
I will create a new issue in ol-django to support bulk create/update as well as adding the datetime attribute to the HubSpotObject.  This will ensure that no other apps need to call HubSpot directly and can instead rely on the ol-django methods.